### PR TITLE
Add ability to scan for new readers and also delete them

### PR DESCRIPTION
**Pull request by kaysabelle**
Make sure you read all the code before approving for merge!

### CODE REVIEW

File: radlibDemoApp/www/index.html [hunk 1]
```diff
@@ -69,20 +79,27 @@ <h1 class="topcoat-navigation-bar__title">RadLib Demo</h1>
...
             <div id="user_input_buttons">
-               <select id="reader_selector" class="topcoat-select">
+               <select id="readersDB" class="topcoat-select controls">
+                  <option value="prompt" selected="true" disabled>Tap here to select an option</option>
+               </select>
+               <!--select id="reader_selector" class="topcoat-select">
                   <option value="prompt" selected="true" disabled>Tap here to select an option</option>
                   <option value="scan">Scan for readers</option>
                   <option value="tsl_1128">TSL 1128</option>
                   <option value="rc522_hf">RC522 HF</option>
```
Comment (jeslin) [inline radlibDemoApp/www/index.html]: commented out code alert!

File: radlibDemoApp/www/index.html [hunk 2]
```diff
@@ -69,20 +79,27 @@ <h1 class="topcoat-navigation-bar__title">RadLib Demo</h1>
...
-               </select>
+               </select-->
                <!--button id="button_stream" class="topcoat-button--cta" onclick="getStream()">Get I/O Stream</button-->
```
Comment (jeslin) [inline radlibDemoApp/www/index.html]: here too

File: radlibDemoApp/www/index.html [hunk 3]
```diff
@@ -98,37 +115,64 @@ <h2 id="status" class="log">Select a reader to get started.</h2>
...
+
          <!-- Katrina's testing buttons -->
          <!--button id="testing" class="testonly" onClick="testpop()">Testing</button>
          <button id="testing2" class="testonly" onClick="testpop2()">Testing2</button-->
 
          <!-- Jeslin's testing buttons -->
-         <!--select id="readersDB" class="topcoat-select">
-            <option value="prompt" selected="true" disabled>Select a Reader</option>
+         <!--select id="readersDB" class="topcoat-select controls">
+            <option value="prompt" selected="true" disabled>Tap here to select an option</option>
```
Comment (jeslin) [inline radlibDemoApp/www/index.html]: more comments!

File: radlibDemoApp/www/js/database.js [hunk 4]
```diff
@@ -88,10 +88,11 @@ function db_checkReaderEntries(object) {
       tx.executeSql("SELECT * FROM READERS WHERE friendlyName= '" + object.friendlyName + "'", [], function (tx, results) {
       // if no results are found
       if (results.rows.length == 0) {
+         alert("new reader!");
          db_addReaderEntry(object);
       }
       else {
-         alert("Reader Name " + object.friendlyName + " already exists!");
+         console.log("Reader Name " + object.friendlyName + " already exists!");
```
Comment (jeslin) [inline radlibDemoApp/www/js/database.js:94]: [insert alert vs console.log debate here]

File: radlibDemoApp/www/js/index.js [hunk 5]
```diff
@@ -26,60 +32,105 @@ $('#cheque').change(
...
+function selectConnectionScreen() {
+   $('.controls').hide();
+   $('.selectConnectionscan').show();
+   $('#tagsDB').hide();
+   dumpLog("Please choose a connection type.");
+}
+
+function selectConnectionType() {
+   var selectConnectButton = $('#button_selectconnection');
+   //$(selectConnectButton).prop("disabled", true);
```
Comment (jeslin) [inline radlibDemoApp/www/js/index.js]: comment

File: radlibDemoApp/www/js/index.js [hunk 6]
```diff
@@ -26,60 +32,105 @@ $('#cheque').change(
...
    else {
-      alert("Please select a reader");
+      dumpLog("Please select a valid connection type");
    }
 }
 
 //sample function to show how java returns the device names/addresses
-function deleteMe(data){
+// DO NOT DELETE!!! MUST RENAME LATER!!!!!!!
+function deleteMe(data) {
```
Comment (jeslin) [inline radlibDemoApp/www/js/index.js]: RENAME RENAME RENAME!

File: radlibDemoApp/www/js/index.js [hunk 7]
```diff
@@ -26,60 +32,105 @@ $('#cheque').change(
...
-      string += data[i].name + " " + data[i].address + "\n";
+   for(var i = 0; i < data.length;i++) {
+      //string += data[i].name + " " + data[i].address + "\n";
+      document.getElementById("detected_reader_selector").innerHTML += "<option value='" + data[i].address +"'>" + data[i].name + " - " + data[i].address + "</option>";
+   }
+   //alert(string);
+}
+
+function saveReader() {
+   // stop discovery, MOVE THIS LATER
```
Comment (jeslin) [inline radlibDemoApp/www/js/index.js]: heheh comment

File: radlibDemoApp/www/js/index.js [hunk 8]
```diff
@@ -26,60 +32,97 @@ $('#cheque').change(
...
-  objTSL1128.friendlyName = "Friendly UHF Reader Name";
+  objTSL1128.friendlyName = "TSL 1128";
 
 var objRC522 = {};
-  objRC522.connectionType = "BLUETOOTH";
+  objRC522.connection = "BLUETOOTH";
   objRC522.model = "ARDUINO_RC522_HF";
   objRC522.address = "00:14:03:02:03:26";
-  objRC522.friendlyName = "Friendly HF Reader Name";
+  objRC522.friendlyName = "RC522 LF";
```
Comment (MinickLS) [inline radlibDemoApp/www/js/index.js]: this says LF!

Comment (kaysabelle) [inline radlibDemoApp/www/js/index.js]: FIXED :hammer: :nail_care: :nail_care: :nail_care:

File: radlibDemoApp/www/js/database.js [hunk 9]
```diff
@@ -88,10 +88,11 @@ function db_checkReaderEntries(object) {
       tx.executeSql("SELECT * FROM READERS WHERE friendlyName= '" + object.friendlyName + "'", [], function (tx, results) {
       // if no results are found
       if (results.rows.length == 0) {
+         alert("new reader!");
```
Comment (kaysabelle) [inline radlibDemoApp/www/js/database.js]: REMOVE THIS

**Pull request merged**